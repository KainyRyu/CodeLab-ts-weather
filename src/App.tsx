import './App.css';
import Styled from 'styled-components';
// import Main from './pages/Main';
import Main from './pages/MainPage';
import List from './components/List';
import { QueryClient, QueryClientProvider } from 'react-query';

// - 지역 리스트는 hard coding
//   - 지역이름, 좌표값
// - 페이지 랜딩시 현 시간, 특정 지역 날씨 정보 보여주기
// - 랜딩페이지 -> 지역 리스트 클릭 했을 때 날씨 정보 갱신
// - 지역 추가
//   - 각자의 방식으로 저장
// - 지역 삭제

function App() {
  const queryClient = new QueryClient();
  return (
    <div className="App">
      <QueryClientProvider client={queryClient}>
        <Flex>
          <Main />
          <List />
        </Flex>
      </QueryClientProvider>
    </div>
  );
}

const Flex = Styled.div`
  display: flex;
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  justify-content: space-around;
  font-size: calc(10px + 2vmin);
  color: white;
`;

export default App;
