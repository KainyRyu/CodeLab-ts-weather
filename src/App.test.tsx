import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import App from './App';

describe('<App />', () => {
  it('renders component correctly', () => {
    const { container } = render(<App />);

    expect(container).toMatchSnapshot();
  });
});
