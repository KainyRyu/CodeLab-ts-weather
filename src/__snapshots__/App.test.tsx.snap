// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders component correctly 1`] = `
.c2 {
  width: 300px;
  height: 300px;
  border: white 1px solid;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #282c34;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  font-size: calc(10px + 2vmin);
  color: white;
}

<div>
  <div
    class="App"
  >
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <div>
          Here, Now
        </div>
        <div
          class="c2"
        >
          <div>
            London
          </div>
          <div>
            17.24
            â„ƒ
          </div>
          <div>
            5.66
            m/s
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
